spec/lib/direct_message_channel_creator_spec.rb

```diff
@@ -106,6 +108,33 @@
       }.to not_change { ChatChannel.count }.and change { UserChatChannelMembership.count }.by(1)
       expect(existing_channel).to eq(own_chat_channel)
     end
+
+    context "when the user is not a member of direct_message_enabled_groups" do
+      before { SiteSetting.direct_message_enabled_groups = Group::AUTO_GROUPS[:trust_level_4] }
+
+      it "raises an error and does not change membership or channel counts" do
+        channel_count = ChatChannel.count
+        membership_count = UserChatChannelMembership.count
+        expect {
+          existing_channel = subject.create!(acting_user: user_1, target_users: [user_1, user_1])
+        }.to raise_error(Discourse::InvalidAccess)
+        expect(ChatChannel.count).to eq(channel_count)
+        expect(UserChatChannelMembership.count).to eq(membership_count)
+      end
+
+      context "when user is staff" do
+        before { user_1.update!(admin: true) }
+
+        it "doesn't create an error and doesn't create a new chat channel" do
```

Comment:
I'm not sure I understand why an admin is not allowed to create a new chat channel here? Since the setting is configured to TL4, it means anyone TL4 and above can create a direct message right?

Reply:
This is returning the existing channel based on the context above, will slightly change this description so it is clearer.